lib/mapObjects/CGTownInstance.cpp

```diff
@@ -487,72 +503,73 @@ void CGTownInstance::newTurn(CRandomGenerator & rand) const
 			cb->setObjProperty (id, ObjProperty::BONUS_VALUE_FIRST, resID);
 			cb->setObjProperty (id, ObjProperty::BONUS_VALUE_SECOND, resVal);
 		}
-
-		const auto * manaVortex = getBonusingBuilding(BuildingSubID::MANA_VORTEX);
-
-		if (manaVortex != nullptr)
+		
+		for(const auto * manaVortex : getBonusingBuildings(BuildingSubID::MANA_VORTEX))
 			cb->setObjProperty(id, ObjProperty::STRUCTURE_CLEAR_VISITORS, manaVortex->indexOnTV); //reset visitors for Mana Vortex
-
+		
```

Comment:
here

Reply:
ok, but those added by xcode so I have to revert them line by line manually